otwcode/otwarchive
# AO3-5750 Upgrade factory_girl to factory_bot

# Pull Request Checklist


## Issue

https://otwarchive.atlassian.net/browse/AO3-5750

## Purpose

What does this PR do?

* Replaces all uses of static attributes in factories with dynamic attributes, required for v4.11.0 onwards: https://thoughtbot.com/blog/deprecating-static-attributes-in-factory_bot-4-11.
* Installs the gem `factory_bot` instead of its now-deprecated predecessor `factory_girl`.
* Updates the `pickle` gem, since otherwise creating users in cucumber features breaks.
* Updates the bundled major version of factory_bot to 5.x

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended? (If you have access, please copy this into the JIRA ticket for them!)

This is a tests-only change, so there is nothing to test on the site itself.

## References

Are there any other relevant issues / PRs / mailing lists discussions related to this?

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

Enigel

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

she/her
